### PR TITLE
Remember settings of last created game

### LINKED ISSUE #1069: Creating a new game: remembering last options used
Hi,

Usually, when I'm on a server and playtesting many games with a specific deck and format, I need to retype the room details, such as name, format options, spectators can chat, etc. Some way to remember the last options used would be a great help at saving time.

Leaving the room open not always lead to new players coming in, as they usually look for the rooms opened in the last minutes, and ignore rooms open for more than 10 minutes or so.

### PR DESCRIPTION
**Pull request by Fizztastic**
Fix #1069 

-Adds a reset button to the create game dialog
reset

### CODE REVIEW

**Comment by ZeldaZach** [comment]
Can you attach a picture please?

File: cockatrice/src/settingscache.cpp [hunk 1]
```diff
@@ -82,6 +82,25 @@ SettingsCache::SettingsCache()
     masterVolume = settings->value("sound/mastervolume", 100).toInt();
 
     cardInfoViewMode = settings->value("cards/cardinfoviewmode", 0).toInt();
+
+    highlightWords = settings->value("personal/highlightWords", QStringList()).toStringList();
+	
+	gameDescription = settings->value("game/gamedescription","").toString();
```
Comment (ZeldaZach) [inline cockatrice/src/settingscache.cpp]: Can you convert this file to spaces instead of tabs?

File: cockatrice/src/settingscache.cpp [hunk 2]
```diff
@@ -82,6 +82,23 @@ SettingsCache::SettingsCache()
...
+    gamePassword = settings->value("game/gamepassword", "").toString();
+    onlyBuddies = settings->value("game/onlybuddies", false).toBool();
+    onlyRegistered = settings->value("game/onlyregistered", true).toBool();
+    spectatorsAllowed = settings->value("game/spectatorsallowed", true).toBool();
+    spectatorsNeedPassword = settings->value("game/spectatorsneedpassword", false).toBool();
+    spectatorsCanTalk = settings->value("game/spectatorscantalk", false).toBool();
+    spectatorsCanSeeEverything = settings->value("game/spectatorscanseeeverything", false).toBool();
+}
+
+void SettingsCache::setHighlightWords(const QString _highlightWords) {
```
Comment (ZeldaZach) [inline cockatrice/src/settingscache.cpp]: Why is this here? This is from my PR

**Comment by ctrlaltca** [comment]
It seems good to me; the only thing, the gameTypes string concatenation is a bit tricky. I know that QSettings doesn't play nice with QList, so you're forced into concatenating the game types in a single QString.
If you want to avoid the hassle of concatenating the string yourself, you can use ```QString::split()``` in ```getGameTypes()``` and ```QStringList::join()``` in ```setGameTypes()``` to make qt do the dirty job.

**Comment by Daenyth** [comment]
As a user new to this feature, `reset` isn't entirely clear. It's a strange word in this context. I'm not sure what's better

**Comment by poixen** [comment]
Default. Default settings. Return to default. Ect ect

**Comment by Fizztastic** [comment]
Would "Clear" be better? Because normally you either Reset or Clear a form.

**Comment by ZeldaZach** [comment]
Clear seems fine to me

File: cockatrice/src/dlg_creategame.cpp [hunk 3]
```diff
@@ -157,8 +175,41 @@ DlgCreateGame::DlgCreateGame(const ServerInfo_Game &gameInfo, const QMap<int, QS
...
+    }
+
+    descriptionEdit->setFocus();
+}
+
 void DlgCreateGame::actOK()
 {
+    settingsCache->setGameDescription(descriptionEdit->text());
+    settingsCache->setMaxPlayers(maxPlayersEdit->value());
+    settingsCache->setGamePassword(passwordEdit->text());
```
Comment (poixen) [inline cockatrice/src/dlg_creategame.cpp]: Not sure I like the idea of storing plain text passwords. We do it already for logins, which we don't like. Also doesn't feel like it should be stored anyway.

Comment (ZeldaZach) [inline cockatrice/src/dlg_creategame.cpp]: Re-thinking this, you're partially right. It doesn't really matter that the game passwords are stored as plain text (as they have no significance to anything other then who could join a game), but I don't see a reason a user would/should use the same PW for each and every game he's making.

Comment (poixen) [inline cockatrice/src/dlg_creategame.cpp]: The password is even visible. How is this partially right. It's bad design.

Comment (ZeldaZach) [inline cockatrice/src/dlg_creategame.cpp]: It's a game-room password that you're supposed to share. Why shouldn't you be able to see it?

Comment (poixen) [inline cockatrice/src/dlg_creategame.cpp]: Because others can see it if they are in the same location.

Comment (ZeldaZach) [inline cockatrice/src/dlg_creategame.cpp]: But this isn't a password to anything really relevant. It's not an account or anything, it's for a simple game room. Just kick them if they weren't meant to be there. I'm up for a discussion on gitter if you'd like

Comment (poixen) [inline cockatrice/src/dlg_creategame.cpp]: You should not store any passwords without the users consent. End-of.

**Comment by woogerboy21** [comment]
@poixen correct, storing passwords in clear text regardless of importance is bad practice.

**Comment by ZeldaZach** [comment]
@woogerboy21 Thanks for the additional input.

@Fizztastic Can you remove the storage of that field from the PR please?

**Comment by ZeldaZach** [comment]
Seems good to me. :shipit:

**Comment by ZeldaZach** [comment]
@Fizztastic Can you fix the merge conflicts? Once that's done, I'll be able to merge :smile:

**Comment by Daenyth** [comment]
And fix the commit history too (rebase/squash/rewrite commit message)

**Comment by ZeldaZach** [comment]
It was a cluster mess, I rebased/did it for him
